src/phoenix/server/api/mutations/user_mutations.py

```diff
@@ -51,34 +82,133 @@ async def create_user(
+        return UserMutationPayload(
+            user=User(
+                id_attr=user.id,
+                email=user.email,
+                username=user.username,
+                created_at=user.created_at,
+                user_role_id=user.user_role_id,
+            )
+        )
+
+    @strawberry.mutation(
+        permission_classes=[
+            IsNotReadOnly,
+            HasSecret,
+            IsAuthenticated,
+            IsAdmin,
+        ]
+    )  # type: ignore
+    async def patch_user(
+        self,
+        info: Info[Context, None],
+        input: PatchUserInput,
+    ) -> UserMutationPayload:
+        user_id = from_global_id_with_expected_type(input.user_id, expected_type_name=User.__name__)
+        async with AsyncExitStack() as stack:
+            session = await stack.enter_async_context(info.context.db())
+            if not (user := await session.scalar(_select_user_by_id(user_id))):
+                raise ValueError("User not found")
+            stack.enter_context(session.no_autoflush)
+            if input.role:
+                user_role_id = await session.scalar(
+                    select(models.UserRole.id).where(models.UserRole.name == input.role.value)
                 )
-            assert user is not None
-        except IntegrityError as error:
-            raise ValueError(_get_user_create_error_message(error))
+                if user_role_id is None:
+                    raise ValueError(f"Role {input.role.value} not found")
+                user.user_role_id = user_role_id
+            if password := input.password:
+                if user.auth_method != enums.AuthMethod.LOCAL.value:
+                    raise ValueError("Cannot modify password for non-local user")
+                validate_password_format(password)
+                user.password_salt = secrets.token_bytes(DEFAULT_SECRET_LENGTH)
+                user.password_hash = await info.context.hash_password(password, user.password_salt)
+                user.reset_password = True
+            if username := input.username:
+                user.username = username
+            assert user in session.dirty
+            try:
+                await session.flush()
+            except IntegrityError as error:
+                raise ValueError(_user_crud_error_message(error, "modify"))
+        assert user
+        if input.password:
+            await info.context.log_out(user.id)
+            await info.context.log_in(user)
+        return UserMutationPayload(
+            user=User(
+                id_attr=user.id,
+                email=user.email,
+                username=user.username,
+                created_at=user.created_at,
+                user_role_id=user.user_role_id,
+            )
+        )
+
+    @strawberry.mutation(
+        permission_classes=[
+            IsNotReadOnly,
+            HasSecret,
+            IsAuthenticated,
+        ]
+    )  # type: ignore
+    async def patch_self_user(
```

Comment:
There's already the term Viewer in the schema. let's call it patch_viewer

Reply:
ok